patcher.sh

```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/env bash
+
+# exit on any error to avoid showing everything was successfull even though it wasnt
+set -eou pipefail
+
+# Patching user variables
+OSCHOICE=""
+AM2RZIP=""
+HQMUSIC=false
+SYSTEMWIDE=false
+APPIMAGE=false
+PREFIX=""
+
+# Patching internal variables
+# Since people are likely to double click on this, I need a way to get the script_dir
+dirResult="$(dirname '${BASH_SOURCE[0]}')"
+SCRIPT_DIR="$(realpath $dirResult)"
+VERSION="1.5.5"
+OUTPUT="${SCRIPT_DIR}/am2r_${VERSION}"
+RESOURCES="${OUTPUT}/assets"
+
+patch_am2r ()
+{
+	# Set prefix to default value if empty
+	if [ -z "$PREFIX" ]; then
+		if [ "$SYSTEMWIDE" = true ]; then
+			PREFIX="/usr/local"
+		else
+			PREFIX="$SCRIPT_DIR"
+		fi
+	fi
+
+	# We need to do variable adjustments based on the prefix
+	if [ "$SYSTEMWIDE" = true ]; then
+		OUTPUT="${PREFIX}/opt/am2r"
+		RESOURCES="${OUTPUT}/assets"
+	else
+		OUTPUT="${PREFIX}/am2r_${VERSION}"
+		RESOURCES="${OUTPUT}/assets"
+	fi
+
+	# Cleanup in case the dirs/files exists
+	rm -rf "$OUTPUT" "$SCRIPT_DIR/AM2R.AppDir" "$SCRIPT_DIR/AM2R-x86_64.AppImage" "$SCRIPT_DIR/AM2RWrapper/" "$SCRIPT_DIR/utilities/android/assets/" "$SCRIPT_DIR/AM2RWrapper.apk"
+
+	# Create necessary directories
+	mkdir -p "$OUTPUT" "$RESOURCES"
+
+	# Check for AM2R_11
+	if [[ -f "$AM2RZIP" ]]; then
+		echo "AM2R_11.zip found!"
+		unzip -q "$AM2RZIP" -d "$OUTPUT"
+	elif [[ -d "$AM2RZIP" ]]; then
+		echo "AM2R_11 folder found!"
+		cp -R "$AM2RZIP" "$OUTPUT"
+	else
+		echo "AM2R_11 not found! Please place AM2R_11.zip (case sensitive) into \"$SCRIPT_DIR\" or provide it via command line arguments and try again."
+		return 1
+	fi
+
+	# Check for which OS we patch
+	if [ "$OSCHOICE" = "linux" ]; then
+
+		# Check whether Xdelta is installed
+		if [ ! command -v xdelta3 &> /dev/null ] ; then
+			>&2 echo "Xdelta is not installed! Please install 'xdelta3' from your local package manager!"
+			return 1
```

Comment:
Just to communicated this is a fatal problem a bit clearer:
```suggestion
			exit 1
```

Reply:
Done locally.